clients/ops/admin-ui/src/features/datastore-connections/datastore-connection.slice.ts

```diff
@@ -232,6 +244,25 @@ export const datastoreConnectionApi = createApi({
         }
       },
     }),
+    patchDataset: build.mutation<any, PatchDatasetsRequest>({
+      query: (params) => ({
+        url: `${CONNECTION_ROUTE}/${params.connection_key}/dataset`,
+        method: "PATCH",
+        body: params.datasets,
```

Comment:
This needs to always be a list in order to conform to the API spec

Frustratingly I can't deeplink you to the OpenAPI spec:
![Screenshot 2022-09-12 at 13 01 49](https://user-images.githubusercontent.com/1667340/189713437-f1740618-10f7-4efa-88d7-984a877c13d3.png)